src/run/config.rs

```diff
@@ -17,12 +18,19 @@ pub struct Config {
     pub skip_setup: bool,
 }
 
+impl Config {
+    pub fn set_token(&mut self, token: Option<String>) {
+        self.token = token;
+    }
+}
+
 #[cfg(test)]
 impl Config {
     /// Constructs a new `Config` with default values for testing purposes
     pub fn test() -> Self {
         Self {
             upload_url: Url::parse(DEFAULT_UPLOAD_URL).unwrap(),
+            frontend_url: Url::parse("https://codspeed.io/").unwrap(),
```

Comment:
The CLI should not be responsible for the front-end base URL. It would be better to return it from the session endpoint directly.

Reply:
It was already done like that for the `createLoginSession` endpoint.
I will add a new resolver on `run` so that we can retrieve the url from the API.